### PR TITLE
EIP: Add DID related methods to the JSON-RPC

### PR DESCRIPTION
**Pull request by oed**
Discussions to: https://github.com/ethereum/EIPs/issues/2845

### CODE REVIEW

File: EIPS/eip-add-did-related-methods.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
+---
+eip: <to be assigned>
```
Comment (MicahZoltu) [inline EIPS/eip-add-did-related-methods.md:2]: Use 2844 for the number, and make sure to update the file name to `eip-2844.md`

File: EIPS/eip-add-did-related-methods.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
...ting signed and encrypted objects. Another shortcoming of the previous approach is that it's impossible to retrieve the `x25519` public key from another user if only an Ethereum address is known. Public key discoverability is at the core of the work that is happening with the [W3C DID standard](https://w3c.github.io/did-core), where given a DID a document which contains public keys can always be discovered. Implementations of this standard already exist and are adopted within the Ethereum community, e.g. [`did:ethr`](https://github.com/decentralized-identity/ethr-did-resolver/) and [`did:3`](https://github.com/3box/3id-resolver). Interoperability between JOSE/COSE and DIDs [already exists](https://github.com/decentralized-identity/did-jwt), and work is being done to [strengthen it](https://github.com/decentralized-identity/did-jose-extensions). Adding support for JOSE/COSE and DIDs will enable Ethereum wallets to support a wide range of new use cases such as more traditional authentication using JWTs, as well as new emerging technologies such as [Secure Data Stores](https://identity.foundation/secure-data-store/) and [encrypted data in IPFS](https://github.com/ipld/specs/pull/269).
```
Comment (MicahZoltu) [inline EIPS/eip-add-did-related-methods.md:24]: Consider mentioning what DID, JOSE, and COSE stand for somewhere in this document, ideally near the first usage.

File: EIPS/eip-add-did-related-methods.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+##### Returns:
+
+* `did` - the DID which authentication was given for
+* `paths` - the paths which was given permission for
+
+
+
+#### CreateJWS
+
+Creates a JSON Web Signature.
```
Comment (MicahZoltu) [inline EIPS/eip-add-did-related-methods.md:52]:
```suggestion
Creates a JSON Web Signature (JWS).
```

File: EIPS/eip-add-did-related-methods.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+* `protected` - the protoected header, json object
+* `did` - the DID that should sign the message, may include the key fragment, string
+
+##### Returns:
+
+A JWS with compact serialization, string
+
+##### Recommendation:
+
+Use `secp256k1` for singning, alternatively `ed25519`.
```
Comment (MicahZoltu) [inline EIPS/eip-add-did-related-methods.md:72]: This seems like it needs to be well standardized as the recipient of the signature needs to know whether it was signed with `secp256k1` or `ed25519`.  Am I incorrect in this?  Can it be deduced which way it was signed without having to probe?

File: EIPS/eip-add-did-related-methods.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+* `protected` - the protoected header, json object
+* `did` - the DID that should sign the message, may include the key fragment, string
+
+##### Returns:
+
+A CWE encoded as base64, string
+
+##### Recommendation:
+
+Use either `secp256k1` for singning.
```
Comment (MicahZoltu) [inline EIPS/eip-add-did-related-methods.md:98]:
```suggestion
Use `secp256k1` for singning.
```

File: EIPS/eip-add-did-related-methods.md (hunk 4 above)
Comment (oed) [reply EIPS/eip-add-did-related-methods.md:72]: The JWS envelope should always contain an `alg` property in the header which specifies which signature algorithm was used.

File: EIPS/eip-add-did-related-methods.md (hunk 2 above)
Comment (oed) [reply EIPS/eip-add-did-related-methods.md:24]: Good suggestion,  thanks!

**Comment by oed** [comment]
Thanks for your feedback @MicahZoltu! Updated the PR with the EIP number and addressed some of your comments.

**Pull request merged**